### PR TITLE
Potential perf improvements to result batching

### PR DESCRIPTION
**Pull request by fbernier**
I'm still new to Go and I don't really know what I'm doing but I'll add actual measurements soon.

### CODE REVIEW

File: rows.go [hunk 1]
```diff
@@ -57,7 +58,9 @@ type rows struct {
 	tzOffset      string
 	inMemRowLimit int
 	resultCache   *os.File
+	ReadWriteBuf  *bufio.ReadWriter
 	cachingFailed bool
+	scratch       [512]byte
```
Comment (atomirex) [inline rows.go]: Any reason it's 512?

Comment (fbernier) [reply rows.go]: No. I don't really have an idea what a good size could be.

**Comment by fbernier** [comment]
Benchmark needs cleanup but

```
+benchcmp master.bench current.bench
benchmark                 old ns/op     new ns/op     delta
BenchmarkBasicQuery-2     53447989      31190472      -41.64%
benchmark                 old allocs     new allocs     delta
BenchmarkBasicQuery-2     145639         138370         -4.99%
benchmark                 old bytes     new bytes     delta
BenchmarkBasicQuery-2     4491280       4392140       -2.21%
```

**Comment by fbernier** [comment]
@huebnerr same as #47

**Comment by fbernier** [comment]
rebased

**Pull request merged**